partiql-parser/src/peg.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright Amazon.com, Inc. or its affiliates.
+
+//! Contains the [Pest](https://pest.rs) defined parser for PartiQL and a wrapper APIs that
+//! can be exported for users to consume.
+
+use crate::prelude::*;
+use pest::Parser;
+use pest_derive::Parser;
+
+#[derive(Parser)]
+#[grammar = "partiql.pest"]
+struct PartiQLParser;
```

Comment:
This is the macro that generates the Rust PEG parser from our declarative definition.  Note that the parser is fairly low-level (returning `Pair` type structures), so we'll never export this directly but wrap it into something that will produce an AST.